src/rgw/driver/rados/rgw_d3n_datacache.cc

```diff
@@ -114,10 +120,17 @@ void D3nDataCache::init(CephContext *_cct) {
 int D3nDataCache::d3n_io_write(bufferlist& bl, unsigned int len, std::string oid)
 {
   D3nChunkDataInfo* chunk_info{nullptr};
-  std::string location = cache_location + url_encode(oid, true);
+  std::string location = cache_location + oid;
   int r = 0;
 
   lsubdout(g_ceph_context, rgw_datacache, 20) << "D3nDataCache: " << __func__ << "(): location=" << location << dendl;
+  std::filesystem::path dir = std::filesystem::path(location).parent_path();
+  std::error_code sec;
+  if (!std::filesystem::exists(dir) && !std::filesystem::create_directories(dir, sec)) {
+    std::cerr << "Error creating directories: " << sec.message() << std::endl;
```

Comment:
i guess this is just breaking up the object name into path segments? which would mean we'd still fail on long object names without slashes, or paths with long segment names

perhaps it would be better to break the names into fixed-length segments, instead of looking for slashes? it seems like that strategy could work for all names, but would still require the url-encoding

Reply:
ack, checking on a 250 chars long objects name still results in `errno=36 ENAMETOOLONG`
```
ERROR: D3nCacheAioWriteRequest::create_io: open file failed, errno=36, location='/mnt/nvme/rgw_datacache/ccad1a8c-d121-4928-b433-d8e8b2cf0234.4227.2__multipart_models--deepseek-ai--DeepSeek
-R1-Distill-Qwen-32B/snapshots/c55788eddf0bd1b33f44518c6e3fec95fe1240c12b70e8b2b9ef0f2e9a058ac9/c55788eddf0bd1b33f44518c6e3fec95fe1240c12b70e8b2b9ef0f2e9a058ac9/C250_long_file_name-11111111
111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111
11111111111111111111111111111111.2~uDA_YtIuJ6np6tf4IspG1RE1HCxNuyr.1'
```
please see next comment...